File: Max Min - GFG/max-min.java
//{ Driver Code Starts
//Initial Template for Java

//Initial Template for Java


import java.io.*;
import java.util.*;


class Array {
	public static void main (String[] args) {
        
        Scanner sc = new Scanner(System.in);
        
        int t = sc.nextInt();
        while(t-- > 0)
        {
            int n = sc.nextInt();
            int a[] = new int[n];
            
            for(int i=0;i<n;i++){
                a[i] = sc.nextInt();
            }
            
            Solution ob = new Solution();
            
            System.out.println(ob.findSum(a,n));
        }
        
	}
}
// } Driver Code Ends


//User function Template for Java

class Solution
{ 
    public static int findSum(int A[],int N) 
    {
        //code here
        int max = A[0];
        int min = A[0];
        
        for(int i =1; i<N; i++)
        {
            max = max < A[i] ? A[i] : max;
            min = min > A[i] ? A[i] : min;
            
        }
        return max +min;
    }
}

Tc": O(n)
SC

File: Max Min - GFG/Max Min.cpp
class Solution {
public:
    static int findSum(int A[], int N) {
        // Initialize max and min with the first element
        int max = A[0];
        int min = A[0];
        
        // Iterate through the array to find max and min
        for (int i = 1; i < N; ++i) {
            if (A[i] > max) {
                max = A[i];
            }
            if (A[i] < min) {
                min = A[i];
            }
        }
        
        // Return the sum of max and min
        return max + min;
    }
};
